src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,281 @@
+                            'dis', 'distutils', 'doctest', 'email', 'encodings', 'ensurepip', 'enum', 'filecmp',
+                            'fileinput', 'fnmatch', 'formatter', 'fractions', 'ftplib', 'functools', 'genericpath',
+                            'getopt', 'getpass', 'gettext', 'glob', 'graphlib', 'gzip', 'hashlib', 'heapq', 'hmac',
+                            'html', 'http', 'idlelib', 'imaplib', 'imghdr', 'imp', 'importlib', 'inspect', 'io',
+                            'ipaddress', 'json', 'keyword', 'lib2to3', 'linecache', 'locale', 'logging', 'lzma',
+                            'mailbox', 'mailcap', 'mimetypes', 'modulefinder', 'msilib', 'multiprocessing', 'netrc',
+                            'nntplib', 'ntpath', 'nturl2path', 'numbers', 'opcode', 'operator', 'optparse', 'os',
+                            'pathlib', 'pdb', 'pickle', 'pickletools', 'pipes', 'pkgutil', 'platform', 'plistlib',
+                            'poplib', 'posixpath', 'pprint', 'profile', 'pstats', 'pty', 'pyclbr', 'pydoc',
+                            'pydoc_data', 'py_compile', 'queue', 'quopri', 'random', 're', 'reprlib', 'rlcompleter',
+                            'runpy', 'sched', 'secrets', 'selectors', 'shelve', 'shlex', 'shutil', 'signal',
+                            'site-packages', 'site', 'smtpd', 'smtplib', 'sndhdr', 'socket', 'socketserver', 'sqlite3',
+                            'sre_compile', 'sre_constants', 'sre_parse', 'ssl', 'stat', 'statistics', 'string',
+                            'stringprep', 'struct', 'subprocess', 'sunau', 'symbol', 'symtable', 'sysconfig',
+                            'tabnanny', 'tarfile', 'telnetlib', 'tempfile', 'test', 'textwrap', 'this', 'threading',
+                            'timeit', 'tkinter', 'token', 'tokenize', 'trace', 'traceback', 'tracemalloc', 'tty',
+                            'turtle', 'turtledemo', 'types', 'typing', 'unittest', 'urllib', 'uu', 'uuid', 'venv',
+                            'warnings', 'wave', 'weakref', 'webbrowser', 'wsgiref', 'xdrlib', 'xml', 'xmlrpc',
+                            'zipapp', 'zipfile', 'zipimport', 'zoneinfo', '_aix_support', '_bootlocale',
+                            '_bootsubprocess', '_collections_abc', '_compat_pickle', '_compression', '_markupbase',
+                            '_osx_support', '_pydecimal', '_pyio', '_py_abc', '_sitebuiltins', '_strptime',
+                            '_threading_local', '_weakrefset', '__future__', '__phello__', '__pycache__']
+        standard_lib.extend(sys.builtin_module_names)
+        standard_lib.append("sas")
+
+        for module_name in sys.modules.keys():
+
+            package_name = module_name.split('.')[0]
+
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if module_name in standard_lib or module_name in package_versions_dict:
+                continue
```

Comment:
I don't know that you need to check for `module_name` separately from `package_name` in you dict. If `module_name` does not contain dot then `package_name` will be `module_name`. Conversely, if `module_name` does contain dot then it won't match anything because you never enter any dotted names in the dict.

Reply:
True